test/clj/rems/browser_test_util.clj

```diff
@@ -248,3 +249,45 @@
 
 (defn wait-for-downloads [string-or-regex]
   (wait-predicate #(seq (downloaded-files string-or-regex))))
+
+(defn- value-of-el
+  "Return the \"value\" an element `el`.
+
+  Value is the first of
+  - read-only checkbox has \"aria-checked\"
+  - input has \"value\",
+  - other (read-only) fields have text
+
+  Mostly we check `el` children in case the structure is deep."
+  [el]
+  (->> [(try ; checkbox
+          (when-let [checkbox-el (child el {:css ".checkbox"})]
+            (= "true" (get-element-attr-el checkbox-el "aria-checked")))
+          (catch Exception _))
+
+        (try ; input with value
+          (when-let [input-el (if (get-element-attr-el el "value")
+                                el
+                                (child el {:css "input"}))]
+            (when-let [v (get-element-attr-el input-el "value")]
+              (when-not (str/blank? v) ; e.g. dropdown doesn't use value
+                (str/trim v))))
+          (catch Exception _))
+
+        ;; others
+        (when-let [v (get-element-text-el el)]
+          (str/trim v))]
+
+       (remove nil?) ; we want false but not nil
+       first))
```

Comment:
I think `(first (remove nil? ...))` is just `(some some? ...)`

Reply:
True but which is more readable for average Clojurist?

Reply:
yeah I agree that `some some?` is a bit cryptic, but they are standard functions after all :wink: 

not insisting you change this

Reply:
actually, let me take that back, `(some some? ...)` doesn't do the same thing because `some?` returns a bool

Reply:
I haven't seen `some some?` in the wild a lot yet.

I had `some some?` and some variants with `nil?`, `or`, `cond`, `remove` etc. there at a time. I think this is slightly complex code but had no better ideas yet.